engine/utils/HeapArray.h

```diff
@@ -515,7 +516,9 @@ class HeapArray
                   masksSize,
                   Memory::Allocate<uint8_t>(BITMASK_SIZE * masksSize),
                   Memory::Allocate<T>(sizeof(T) * newSize))
-    {}
+    {
+        memset(stateMaskBitfield, 0, (BITMASK_SIZE * masksSize));
+    }
```

Comment:
What change is this part of?

Reply:
This is to get the unit tests to pass. There was a memory bug that only linux caught. Basically the statemask wasn't being set to zero after being allocated. This fixes that issue.